benchmark/C/Savina/src/parallelism/PiPrecision.lf

```diff
@@ -196,13 +189,24 @@ reactor Worker {
     input doWork:uint32_t;
     output response:mpf_t;
     
+    reaction(startup) {=
+        mpf_init_set_ui(sixteen, 16ul);
+        mpf_init_set_ui(four, 4ul);
+        mpf_init_set_ui(two, 2ul);
+        mpf_init_set_ui(one, 1ul);
+    =}
+    
     reaction(doWork) -> response {=
         mpf_t result;
         mpf_init(result);
         calculateBbpTerm(result, doWork->value);
         mpf_swap(response->value, result);
         SET_PRESENT(response);
     =}
+    
+    reaction(shutdown) {=
+        // mpf_clears(one, two, four, sixteen, NULL); FIXME: This causes a "double free or corruption" error
```

Comment:
That is not surprising, as this shutdown reaction will be invoked multiple times for each worker. If we use global variables, then we should probably allocate and free them in the main reactor.

Reply:
Ah! You're right. Completely missed that.